test/libsolidity/semanticTests/functionCall/gas_and_value_basic.sol

```diff
@@ -39,9 +39,9 @@ contract test {
 
 // ----
 // constructor(), 20 wei ->
-// gas irOptimized: 262130
-// gas legacy: 402654
-// gas legacyOptimized: 274470
+// gas irOptimized: 303208
+// gas legacy: 443715
+// gas legacyOptimized: 315548
```

Comment:
It's not clear to my why the gas increased for these two tests. They are both deploying a new contract inside a constructor. My initial theory was that in such cases in the past, metadata wasn't appended. But I tried to deploy contracts this way and that is not the case :(

Reply:
Hm... the semantics tests *should* have removed the metadata in general in order not to get flaky gas costs due to zeroes or non-zeroes and such...
```
	if (m_enforceGasCost)
	{
		m_compiler.setMetadataFormat(CompilerStack::MetadataFormat::NoMetadata);
		m_compiler.setMetadataHash(CompilerStack::MetadataHash::None);
	}
```
in ``test/libsolidity/SmenaticTest.cpp``....

Ah, I see now... ``SolidityExecutionFramework::multiSourceCompileContract`` later on calls ``m_compiler.reset()``, which earlier did *not* reset the metadata format, but now it does. That's the reason.

Reply:
i.e. quick fix probably is to add a ``m_metadataFormat`` to ``SolidityExecutionFramework`` and set that instead of invoking ``m_compiler.setMetadataFormat`` in ``SemanticTest.cpp``...
``m_compiler.setMetadataHash`` in ``SemanticTest.cpp`` was silly anyways, since it just got overwritten :-).

At least if I saw correctly just now, only had a quick glance :-).

Reply:
Thanks, that actually worked!

Reply:
Now tests seem to be failing with `Invalid or missing metadata in bytecode.` :)

Reply:
Yeah, have to fix the new issue :(